File: app/src/main/java/dev/gdgaddis/codenightsmsblocker/service/Boot.java
package dev.gdgaddis.codenightsmsblocker.service;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class Boot  extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            Intent serviceIntent = new Intent(context, BlockerService.class);
            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O)
                context.startForegroundService(serviceIntent);
            else
                context.startService(serviceIntent);
        }
    }
}

File: app/src/main/java/dev/eyosiyas/smsblocker/service/Boot.kt
package dev.eyosiyas.smsblocker.service

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build

class Boot : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if ((Intent.ACTION_BOOT_COMPLETED == intent.action)) {
            val serviceIntent = Intent(context, BlockerService::class.java)
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) context.startForegroundService(serviceIntent) else context.startService(serviceIntent)
        }
    }
}